hyphen.c

```diff
@@ -1076,17 +1076,20 @@ int hnj_hyphen_norm(const char *word, int word_size, char * hyphens,
 void hnj_hyphen_hyphword(const char * word, int l, const char * hyphens, 
     char * hyphword, char *** rep, int ** pos, int ** cut)
 {
-  int hyphenslen = l + 5;
+  /* hyphword buffer size must be at least 2 * l */
+  int hyphword_len = 2 * l - 1;
```

Comment:
I believe that it is sane to assume `UINT_MAX > 2 * INT_MAX`. 
But if you prefer we may also include <limits.h> and do an explicit check.